cmd/prow-job-dispatcher/main.go

```diff
@@ -370,4 +424,61 @@ func main() {
 	if err := dispatcher.SaveConfig(config, o.configPath); err != nil {
 		logrus.WithError(err).Fatalf("Failed to save config file to %s", o.configPath)
 	}
+
+	if !o.createPR {
+		logrus.Info("Finished dispatching and create no PR, exiting ...")
+		os.Exit(0)
+	}
+
+	logrus.Infof("Changing working directory to '%s' ...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+
+	steps := []struct {
+		gitOnly   bool
+		command   string
+		arguments []string
+	}{
+		{
+			gitOnly:   true,
+			command:   "/usr/bin/prow-job-dispatcher",
+			arguments: []string{"--prow-jobs-dir", "./ci-operator/jobs", "--prometheus-url", o.PrometheusURL, "--prometheus-password-path", o.PrometheusPasswordPath, "--prometheus-username", o.PrometheusUsername, "--config-path", "core-services/sanitize-prow-jobs/_config.yaml"},
+		},
+		{
+			command:   "/usr/bin/sanitize-prow-jobs",
+			arguments: []string{"--prow-jobs-dir", "./ci-operator/jobs", "--config-path", "./core-services/sanitize-prow-jobs/_config.yaml"},
+		},
+	}
+
+	stdout := bumper.HideSecretsWriter{Delegate: os.Stdout, Censor: sa}
+	stderr := bumper.HideSecretsWriter{Delegate: os.Stderr, Censor: sa}
+	author := fmt.Sprintf("%s <%s>", o.GitAuthorOptions.GitName, o.GitAuthorOptions.GitEmail)
+	commitsCounter := 0
+
+	for _, step := range steps {
+		committed, err := runAndCommitIfNeeded(stdout, stderr, author, step.command, step.arguments, step.gitOnly)
+		if err != nil {
+			logrus.WithError(err).Fatal("failed to run command and commit the changes")
+		}
+
+		if committed {
+			commitsCounter++
+		}
+	}
+	if commitsCounter == 0 {
+		logrus.Info("no new commits, existing ...")
+		os.Exit(0)
+	}
+
+	if err := bumper.GitPush(fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", o.githubLogin, string(sa.GetTokenGenerator(o.PRCreationOptions.GitHubOptions.TokenPath)()), o.githubLogin, githubRepo), remoteBranch, stdout, stderr, ""); err != nil {
+		logrus.WithError(err).Fatal("Failed to push changes.")
+	}
+
+	title := fmt.Sprintf("%s at %s", matchTitle, time.Now().Format(time.RFC1123))
+	// TODO (hongkliu): Remove this if it passes the manual test
+	labelsToAdd := []string{labels.Hold}
```

Comment:
Please do not add code for this. Just test it. Our presubmits need at the very least 5 minutes to finish their testing, that is plenty of time for you to add a `hold` manually

Reply:
Removed.